nexus/src/app/sagas/instance_migrate.rs

```diff
@@ -21,37 +28,142 @@ use uuid::Uuid;
 #[derive(Debug, Deserialize, Serialize)]
 pub struct Params {
     pub serialized_authn: authn::saga::Serialized,
-    pub instance_id: Uuid,
+    pub instance: db::model::Instance,
     pub migrate_params: params::InstanceMigrate,
 }
 
-// instance migrate saga: actions
-
+// The migration saga is similar to the instance creation saga: get a
+// destination sled, allocate a Propolis process on it, and send it a request to
+// initialize via migration, then wait (outside the saga) for this to resolve.
+//
+// Most of the complexity in this saga comes from the fact that during
+// migration, there are two sleds with their own instance runtime states, and
+// both the saga and the work that happen after it have to specify carefully
+// which of the two participating VMMs is actually running the VM once the
+// migration is over.
+//
+// At the start of the saga, the participating sleds have the following
+// information about the instance's location. (Instance runtime states include
+// other information, like per-Propolis states, that's not relevant here and
+// is ignored.)
+//
+// | Item         | Source | Dest | CRDB |
+// |--------------|--------|------|------|
+// | Propolis gen | G      | None | G    |
+// | Propolis ID  | P1     | None | P1   |
+// | Sled ID      | S1     | None | S1   |
+// | Dst Prop. ID | None   | None | None |
+// | Migration ID | None   | None | None |
 declare_saga_actions! {
     instance_migrate;
+
+    RESERVE_RESOURCES -> "server_id" {
+        + sim_reserve_sled_resources
+        - sim_release_sled_resources
+    }
+
     ALLOCATE_PROPOLIS_IP -> "dst_propolis_ip" {
         + sim_allocate_propolis_ip
     }
-    MIGRATE_PREP -> "migrate_instance" {
-        + sim_migrate_prep
+
+    // This step sets the migration ID and destination Propolis ID fields in
+    // CRDB by asking the instance's current sled to paste them into its runtime
```

Comment:
"paste them" is throwing me off a bit -- similar to the earlier comment, I am not sure where exactly this state is being set

Reply:
Fixed in 321331e.